### PR TITLE
Fix unfriendly admin databases error message

### LINKED ISSUE #20471: Ugly error state in Admin > Databases
image

When the `GET /api/databases` endpoint returns an error you get a page that makes your eyes bleed.

1. Tweak the endpoint so it fails e.g. #20470 
2. Admin > Databases
3. Eyes bleeding

See also #20469

### PR DESCRIPTION
**Pull request by WiNloSt**
Fixes #20471

### Notes/concerns
- This PR modifies `LoadingAndErrorWrapper` which is used across the app. Is this too risky? Should I just fix the error specified in the issue?
- I contemplated changing `LoadingAndErrorWrapper` since we'll probably end up having to fix the error message like #20471 again somewhere else.

#### Before the fix
image

#### After the fix
image

image

### CODE REVIEW

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ describe("scenarios > admin > databases > list", () => {
 
     cy.visit("/admin/databases");
 
-    cy.wait("@failedGet");
+    cy.wait(["@failedGet", "@failedGet"]);
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:21]: Fix the test always pass even before coding. I'm not entirely sure why there're 2 calls to `api/database`

image

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js [hunk 2]
```diff
@@ -6,21 +6,30 @@ describe("scenarios > admin > databases > list", () => {
...
+        });
+      },
+    ).as("failedGet");
 
     cy.visit("/admin/databases");
 
     cy.wait("@failedGet");
     // Not sure how exactly is this going the be fixed, but we should't show the full error message on the page in any case
-    cy.findByText(errorMessage).should("not.exist");
+    cy.findByText(errorMessage).should("not.be.visible");
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:32]: The fix is that I wrap the error message inside a collapsable UI, so it'll always exist, just isn't visible.

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js [hunk 3]
```diff
@@ -1,26 +1,62 @@
...
+    it("should not display error messages upon a failed `GET` (metabase#20471)", () => {
+      const errorMessage = "Lorem ipsum dolor sit amet, consectetur adip";
+
+      cy.intercept(
+        {
+          method: "GET",
+          pathname: "/api/database",
+          query: {
+            exclude_uneditable_details: "true",
+          },
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:44]: Intercept different requests based on whether the version is Enterprise or not. This is needed after https://github.com/metabase/metabase/pull/20953 is merged

File: frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js [hunk 4]
```diff
@@ -1,26 +1,62 @@
...
+            body: { message: errorMessage },
+          });
+        },
+      ).as("failedGet");
+
+      cy.visit("/admin/databases");
+
+      cy.wait("@failedGet");
+      // Not sure how exactly is this going the be fixed, but we should't show the full error message on the page in any case
+      cy.findByText(errorMessage).should("not.be.visible");
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/admin/databases/list.cy.spec.js:30]: The fix is that I wrap the error message inside a collapsable UI, so it'll always exist, just isn't visible.

File: frontend/src/metabase/components/LoadingAndErrorWrapper.jsx [hunk 5]
```diff
@@ -36,6 +37,24 @@ export default class LoadingAndErrorWrapper extends Component {
     messageInterval: 6000,
   };
 
+  renderError(contentClassName) {
+    if (this.props.renderError) {
+      return (
+        <div className="py4">
+          {this.props.renderError(this.getErrorMessage())}
+        </div>
```
Comment (WiNloSt) [inline frontend/src/metabase/components/LoadingAndErrorWrapper.jsx:45]: Avoid wrapping the error in side `contentClassName` and `.text-normal.text-light.ie-wrap-content-fix` which would make the error looks off.

#### Without the wrappers (this PR)
image

#### With the wrappers
image

File: frontend/src/metabase/components/LoadingAndGenericErrorWrapper.jsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import LoadingAndErrorWrapper from "./LoadingAndErrorWrapper";
+
+export default function LoadingAndGenericErrorWrapper(props) {
+  return (
+    <LoadingAndErrorWrapper
+      {...props}
+      renderError={details => <GenericError details={details} />}
+    />
+  );
```
Comment (WiNloSt) [inline frontend/src/metabase/components/LoadingAndGenericErrorWrapper.jsx:13]: Thank @kulyk for [the suggestion on Slack](https://metaboat.slack.com/archives/C505ZNNH4/p1649776410749209?thread_ts=1649773985.732709&cid=C505ZNNH4).

**Review by kulyk — APPROVED** [review]
Nice!

**Comment by kulyk** [comment]
@WiNloSt should we backport it?

**Pull request merged**